### ISSUE TITLE
hotkey to move to next/previous error should not move NVDA's focus outside the editor

### ISSUE THREAD

**Issue by Neurrone** [microsoft/vscode#41356]
<ul>
	<li>VSCode Version: Code - Insiders 1.20.0-insider (f4ed56b0d61050f6a94cfa033a058fabfdc0d651, 2018-01-09T05:13:28.256Z)</li>
	<li>OS Version: Windows_NT x64 10.0.15063</li>
	<li><details><summary>Extensions (2)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-markdownlint|Dav|0.12.0
markdown-preview-enhanced|shd|0.3.2

</details></li>
</ul>

Steps to Reproduce:

1. Create a file that has a lint error. The example I used was:

```markdown
# Heading 1

* HTML
- LaTeX

## heading with trailing space  
```

2. Save the file as "test.md". With NVDA running and vscode-markdownlint, press f8 or shift+f8.

*Expected result:*

The error/warning is read. Keyboard focus stays in the editor, and I move to the offending line.

*Actual result:*

Keyboard focus exits the text area and lands on the text of the error. For e.g, NVDA would say 

```
[markdownlint] MD004/ul-style: Unordered list style [Expected: asterisk; Actual: dash]
```

Focus is now on the text of the alert. I can quickly move to the edit area and press enter to enter forms mode, and focus is placed on the originating line. 

It would be neat if the focus stayed inside the editor though to save this extra step.

**Comment by isidorn** [comment]
This works fine for me on macOS.
The focus stays in the editor and the error is read out.

@Neurrone can you still reproduce this with latest vscode?

**Comment by Neurrone** [comment]
@isidorn yup, this is still happening. If I press f8, my focus moves outside the textarea to where the error message is.

**Comment by isidorn** [comment]
@Neurrone aha, only when the screen reader is detected we actuall move focus.
Would you prefer that we read out the content of the error using an `aria.alert`.
Since currently we move focus only so the screen reader would read the content without us using `aria.alert`.
Or do you have some other recommendation?

**Comment by Neurrone** [comment]
@isidorn yes, having it use alerts would be ideal.

However, when focus moves to the source of the error, I don't know if that is currently being announced. I.e, the alert might interupt the focus movement announcement, if one is present.

**Comment by isidorn** [comment]
Currently it would read the word where it navigates to.
I have changed that now we use aria.alerts.
Can you please try vscode insiders tomorrow and let me know how it behaves? And if you like it or not.
Thank you very much for your help

fyi @jrieken

**Comment by Neurrone** [comment]
@isidorn I've just tested with latest build.

When pressing F8 to move to the next error for a markdown file, this is what I'm hearing:

```
MD032/blanks-around-lists: Lists should be surrounded by blank lines (occurred 2 times) 
alert  MD032/blanks-around-lists: Lists should be surrounded by blank lines
```

The reading of the first sentance is interupted with the alert.

Would be ideal if we could suppress the redundant announcement.

**Comment by isidorn** [comment]
@Neurrone thanks for your feedback.
I have pushed a fix for this. You can try it out tomorrow.
I verified and now it behaves well for me.

We are no longer using an alert, but just aria-livereagion which just reads out the content without stealing focus. So seems to work nice from my persepctive.

**Comment by Neurrone** [comment]
@isidorn thanks a lot, this is working perfectly now :)

**Comment by isidorn** [comment]
Glad to hear, adding verified label
